src/probnum/filtsmooth/gaussfiltsmooth/extendedkalman.py

```diff
@@ -19,8 +19,13 @@ def __init__(self, non_linear_sde, num_steps):
         self.non_linear_sde = non_linear_sde
         self.num_steps = num_steps
 
-    def transition_realization(self, real, start, stop, **kwargs):
-        jacobfun = functools.partial(self.non_linear_sde.jacobian, state=real)
+    def transition_realization(self, real, start, stop, linearise_at=None, **kwargs):
```

Comment:
So `linearise_at` is a RV here, right? Since `real` is a np.ndarray, would it not make sense to have `linearise_at` to be a realization, too?

This could also collapse the code to
```python
jacobfun = functools.partial(self.non_linear_sde.jacobian, state=linearize_at if linearize_at is not None else real)
```
or even
```python
jacobfun = functools.partial(self.non_linear_sde.jacobian, state=linearize_at or real)
```
But if that's better is very subjective of course^^

Reply:
linearise_at is a RV, which I think makes sense even in the transition_realization sense: the UKF for instance, linearises at the full gaussian RV. Or am I wrong here? 

The collapsing would work either way I guess:
```
jacobfun = functools.partial(self.non_linear_sde.jacobian, state=linearize_at.mean or real)
```
but personally I am not sure whether this makes it more readable or only shorter...

Reply:
Yes if the UKF uses the full distribution it makes sense for `linearise_at` to be a RV. How about naming it `linearize_at_rv`? 

`state=linearize_at.mean or real` will fail though if `linearize_at is None` ;) I don't care too much about the if-else. I think I'd prefer removing it (maybe in favor of assigning a variable and then having a single call to `functools.partial`), but I understand if you prefer the readability with the if-else so feel free to keep it as is and just resolve.

Reply:
I think you have a good point with the single call to functools. I will pick a winner tomorrow :) I also think that the `_rv` think might be resolved more cleanly with a type hint? (Since midterm we want to use them anyway...). I will receonsider this thing tomorrow and then resolve if I consider it solved. Thanks for the feedback!